plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -529,6 +529,36 @@ public void testMultipleEqualityDeletesWithDifferentSchemas()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    void testRowTypeWithoutField()
+    {
+        try (TestTable table = newTrinoTable("test_row_without_field", "(id int)")) {
+            assertQueryReturnsEmptyResult("SELECT * FROM " + table.getName());
+            assertQuery("SHOW COLUMNS FROM " + table.getName(), "VALUES ('id', 'integer', '', '')");
+
+            Table icebergTable = loadTable(table.getName());
+            icebergTable.updateSchema().addColumn("nested", Types.StructType.of()).commit();
+
+            // struct type with emtpy fields is ignored
+            assertQueryReturnsEmptyResult("SELECT * FROM " + table.getName());
+            assertQuery("SHOW COLUMNS FROM " + table.getName(), "VALUES ('id', 'integer', '', '')");
+
+            // insert works correctly
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (1)", 1);
```

Comment:
What happens if the column has a default value? Iceberg is going to support the feature in upcoming 1.9.0 as you may know. https://iceberg.apache.org/spec/#default-values

Reply:
Thanks for sharing, it means Iceberg since 1.9.0, column would have [default-values](https://iceberg.apache.org/spec/#default-values).
The current implementation just skips the such empty struct column, no matter it has `default-values` definition in the future. In other words, Trino can't see such column.

If Iceberg allows such empty struct column has `default-values`, the issue is whether we should fully skip handling an empty struct column, or only skip its read/write while still respecting the `default-values` setting during writes if a default value is configured

Reply:
I’m raising this issue from the perspective of other query engines. Users may face unexpected results if we don't handle default values during our write operations.

I’m inclined to cancel this change. Throwing an exception is safer than risking correctness issues.

Reply:
Yeah, understand. 
I am also in favor of of not apply this change. it's should fully support not to skip.